src/components/Admin/AdminTabs.tsx

```diff
@@ -0,0 +1,93 @@
+import { Tabs } from 'flowbite-react'
+import { FaCheckCircle, FaExclamationTriangle, FaTag } from 'react-icons/fa'
+import { IResource } from '@/types'
+import ApprovedCard from '../Resources/ApprovedCard'
+import UnaprovedCard from '../Resources/UnapprovedCard'
+import { useQuery } from 'react-query'
+import { defaultStaleTime, getResourcesWithTags } from '@/api/api'
+import LoadingPage from '../States/LoadingPage'
+import ErrorPage from '../States/ErrorPage'
+
+const tabsTheme = {
+    base: 'flex flex-col gap-2',
+    tablist: {
+        base: 'flex text-center',
+        styles: {
+            default:
+                'flex-wrap -mb-px border-b border-gray-200 dark:border-gray-700',
+        },
+        tabitem: {
+            base: 'flex items-center justify-center p-4 rounded-t-lg text-sm font-medium first:ml-0 disabled:cursor-not-allowed disabled:text-gray-400 disabled:dark:text-gray-500',
+            styles: {
+                default: {
+                    base: 'rounded-t-lg',
+                    active: {
+                        on: 'text-orange-600 rounded-t-lg border-b-2 border-orange-600 active dark:text-orange-500 dark:border-orange-500',
+                        off: 'border-b-2 border-transparent text-gray-500 hover:border-gray-300 hover:text-gray-600 dark:text-gray-400 dark:hover:text-gray-300',
+                    },
+                },
+            },
+            icon: 'mr-2 h-5 w-5',
+        },
+    },
+    tabpanel: 'py-3',
+}
+export default function AdminTabs() {
+    const { isLoading, isError, data } = useQuery(
+        'resources',
+        getResourcesWithTags,
+        {
+            staleTime: defaultStaleTime,
+        }
+    )
+
+    if (isLoading) {
+        return <LoadingPage />
+    }
+    if (isError) {
+        return <ErrorPage />
+    }
+    return (
+        <div className="min-h-screen bg-orange-50 px-12">
+            {data && data.data ? (
```

Comment:
Do we want to use the loading state here? We can also look at a loading wrapper reduce the need to check loading state too often

And if there are no resources, do we want a "no resources found" blurb/state? I think this would result in an infinite loading state if the query doesn't return right?

Reply:
Can you explain more about what you mean by loading wrapper? That sounds like a good option.